packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -109,29 +117,25 @@ export class LambdaInvoke extends sfn.TaskStateBase {
    * @internal
    */
   protected _renderTask(): any {
-    switch (this.props.resourceType) {
-      case 'Arn':
-        return {
-          Resource: this.props.lambdaFunction.functionArn,
-          Parameters: sfn.FieldUtils.renderObject({
-            Payload: this.props.payload ? this.props.payload.value : sfn.TaskInput.fromDataAt('$').value,
-            InvocationType: this.props.invocationType,
-            ClientContext: this.props.clientContext,
-            Qualifier: this.props.qualifier,
-          }),
-        };
-      case 'Invoke':
-      default:
-        return {
-          Resource: integrationResourceArn('lambda', 'invoke', this.integrationPattern),
-          Parameters: sfn.FieldUtils.renderObject({
-            FunctionName: this.props.lambdaFunction.functionArn,
-            Payload: this.props.payload ? this.props.payload.value : sfn.TaskInput.fromDataAt('$').value,
-            InvocationType: this.props.invocationType,
-            ClientContext: this.props.clientContext,
-            Qualifier: this.props.qualifier,
-          }),
-        };
+    if (this.props.payloadResponseOnly) {
+      const res: any = {
+        Resource: this.props.lambdaFunction.functionArn,
+      };
+      if (this.props.payload) {
+        res.Parameters = sfn.FieldUtils.renderObject(this.props.payload.value);
+      }
+      return res;
```

Comment:
can these be combined and simplified a little bit?

```suggestion
      return {
        Resource: this.props.lambdaFunction.functionArn,
        Parameters: sfn.FieldUtils.renderObject({
          Payload: this.props.payload?.value
        }),
      };
```

Reply:
With this form in SFN, `Parameters` is passed to the Lambda API as `Payload`. Unlike the `Payload` parameter in the newer form, you can't use `Parameters.$` and assign it `$` to pass the input undisturbed. Individual properties under `Parameters` can use JSONPath though.

So, if the user specifies a payload, we need to include `Parameters`. If not, we need to not include `Parameters`.

This is more concise and idiomatic.

```suggestion
      return {
        Resource: this.props.lambdaFunction.functionArn,
        ...this.props.payload && { Parameters: sfn.FieldUtils.renderObject(this.props.payload.value); }
      };
```

Reply:
gotcha, I'm in favour of it! it is more succinct :)